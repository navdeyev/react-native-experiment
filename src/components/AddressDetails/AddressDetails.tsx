import * as React from 'react';
import {Text, View} from 'react-native';

import {Address} from '../../domains/types';

interface Props {
    address: Address
}

const AddressDetails: React.SFC<Props> = props => {
    const {address} = props;
    return (
        <View>
            <Text>{address.street}</Text>
            <Text>{address.suite}</Text>
            <Text>{address.city}</Text>
            <Text>{address.zipcode}</Text>
        </View>
    );
};

export default AddressDetails;