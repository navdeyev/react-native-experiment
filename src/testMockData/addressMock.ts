import {Address, Geo} from 'src/domains/types';

export const geo: Geo = {
    lat: '-37.3159',
    lng: '81.1496'
};

export const address: Address = {
    street: 'Kulas Light',
    suite: 'Apt. 556',
    city: 'Gwenborough',
    zipcode: '92998-3874',
    geo
};
