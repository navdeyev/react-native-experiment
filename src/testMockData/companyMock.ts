import {Company} from 'src/domains/types';

export const company: Company = {
    name: 'Romaguera-Crona',
    catchPhrase: 'Multi-layered client-server neural-net',
    bs: 'harness real-time e-markets'
};
