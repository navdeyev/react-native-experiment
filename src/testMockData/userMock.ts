import {User} from '../domains/types';

import {address} from './addressMock';
import {company} from './companyMock';

export const user: User = {
    id: 1,
    name: 'Leanne Graham',
    username: 'Bret',
    email: 'Sincere@april.biz',
    address,
    phone: '1-770-736-8031 x56442',
    website: 'hildegard.org',
    company
};
