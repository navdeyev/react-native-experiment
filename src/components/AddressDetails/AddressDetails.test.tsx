import * as React from 'react';
import {shallow} from 'enzyme';

import AddressDetails from './AddressDetails';
import {address} from 'src/testMockData/addressMock';

describe('AddressDetails', () => {

    it('renders a component', () => {
        const render = shallow(<AddressDetails address={address}/>);
        expect(render.getElement()).toMatchSnapshot();
    });

});
