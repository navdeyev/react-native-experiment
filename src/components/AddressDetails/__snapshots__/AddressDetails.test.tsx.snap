// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressDetails renders a component 1`] = `
<Component>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Kulas Light
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Apt. 556
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Gwenborough
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    92998-3874
  </Text>
</Component>
`;
